test/cpp/common/alarm_test.cc

```diff
@@ -438,6 +439,34 @@ TEST(AlarmTest, UnsetDestruction) {
   Alarm alarm;
 }
 
+TEST(AlarmTest, CallbackSetInCallback) {
+  auto c = std::make_shared<Completion>();
+  struct GuardedAlarm {
+    grpc::internal::Mutex mu;
+    Alarm alarm ABSL_GUARDED_BY(mu);
+  } ga;
+  {
+    grpc::internal::MutexLock l(&ga.mu);
+    ga.alarm.Set(std::chrono::system_clock::now() + std::chrono::seconds(1),
+                 [&](bool ok) {
```

Comment:
If you're capturing everything by reference, you don't need to create a struct to wrap the mutex and alarm. Likely better to just put `alarm` and `alarm_mu` at function scope.

Reply:
done.